flink-core/src/main/java/org/apache/flink/api/common/eventtime/CombinedWatermarkStatus.java

```diff
@@ -69,7 +69,6 @@ public boolean updateCombinedWatermark() {
         // if we don't have any outputs minimumOverAllOutputs is not valid, it's still
         // at its initial Long.MAX_VALUE state and we must not emit that
         if (partialWatermarks.isEmpty()) {
-            this.idle = combinedWatermark > Long.MIN_VALUE;
```

Comment:
Do we want to provide some default Watermark strategy to mimic this behavior? This can be probably done through a follow-up.

Reply:
Good question; I'm also not sure if it's trivial to implement on a higher level... I'm almost assuming that source implementors need to implement it themselves. But afaik we also don't encourage it. @dawidwys WDYT?